src/wallet/db.h

```diff
@@ -35,17 +35,18 @@ class CDBEnv
     void EnvShutdown();
 
 public:
-    mutable CCriticalSection cs_db;
```

Comment:
Why make the lock global instead of per-env? It seems like there should be nothing preventing you from keeping that and adding a second lock that only exists to protect g_dbenvs, though likely not a big deal in terms of concurrency.

Reply:
It just seems weird given it'd be even less code change to not split the lock out, and as long as the global lock is *only* held in the creation of new envs, no lockorder should be worried about (cause lockorder is always trivially correct if locks are held only for the duration of a function which uses them, without making callbacks).